nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'this is need to be implemented.'
+  end
```

Comment:
- [ OPTIONAL ] You've done an awesome job by implementing a method that raises a custom error message instead of a generic "not implemented" one! 🙌 But to take things to the next level and offer an even better user experience, consider including information about the class and method that triggered the error. 🚀
- Picture this: 🎨 You're the tour guide of a magnificent museum, and one of the rooms is temporarily closed for renovations. If someone tries to enter, it's great to have a sign that says, "Room under construction, guided tours not available" instead of just "Closed." But, to make it even more helpful, you can add, "Apologies, the 'Art Gallery' room is closed for renovations. The 'enterRoom' method caused this." 🖼️
- This way, visitors not only know why the room is closed but also which part of the museum's system led to it. It's like giving them a map to navigate the museum of your code! 🗺️

```suggestion
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
```